modules/apps/27-interchain-accounts/types/account.go

```diff
@@ -20,9 +20,10 @@ const (
 	ICAPrefix string = "ics-27"
 )
 
-// GenerateAddress returns a truncated SHA256 hash using the provided port string
-func GenerateAddress(port string) []byte {
-	return tmhash.SumTruncated([]byte(port))
+// GenerateAddress returns an sdk.AccAddress using the provided port identifier and derived from the IBC interchainaccounts module account
```

Comment:
@colin-axner I've included this as per #303 after checking out the comment you had linked! I'm assuming this is a new approach which has yet to be adopted across the sdk. This seems to me like the correct usage but I'm not 100% as other modules use a different pattern.

For example `transfer` is composed with an `authtypes.AccountKeeper` and accesses the module accounts via that interface which from what I can tell are populated through [app.go maccPerms](https://github.com/cosmos/gaia/blob/7a1b89bc138cb6f24b13b78118de0fc928967f4e/app/app.go#L132)

Reply:
Also updated this return sig from `[]byte` to `sdk.AccAddress` as it's probably preferred to use that type for bech32 encoding

Reply:
Nice

Reply:
Nice! Deriving the address looks correct, but we probably need to create the module account in the ICA keeper (in NewKeeper function) and create the sub module account when the address is being generated. I may be mistaken, but typically you have to create a new account in the account keeper since it needs to track account information like sequence

It might be worth doing this in a separate pr. Trying to use this account should test whether everything was setup correctly

Reply:
Sure I can do this in a separate PR 👍 

In the `NewKeeper` func for `transfer` I see assertion that the module account exists:

```

```